client/task-list/tasks/payments/index.js

```diff
@@ -73,21 +73,35 @@ class Payments extends Component {
 			: 'stripe';
 	}
 
-	markConfigured( method, queryParams = {} ) {
-		const { clearTaskStatusCache } = this.props;
+	async setMethodEnabledOption( method, value ) {
+		const { clearTaskStatusCache, updateOptions, options } = this.props;
+		await updateOptions( {
+			[ method.optionName ]: {
+				...options[ method.optionName ],
+				enabled: value,
+			},
+		} );
+
+		clearTaskStatusCache();
+	}
+
+	async markConfigured( methodName, queryParams = {} ) {
 		const { enabledMethods } = this.state;
+		const { methods } = this.props;
+
+		const method = methods.find( ( option ) => option.key === methodName );
 
 		this.setState( {
 			enabledMethods: {
 				...enabledMethods,
-				[ method ]: true,
+				[ methodName ]: true,
 			},
 		} );
 
-		clearTaskStatusCache();
+		await this.setMethodEnabledOption( method, 'yes' );
```

Comment:
markConfigured now persists the option too. I couldn't think of a problem with doing this here.

Reply:
I mentioned a slight problem in the comment above, although an extra condition will fix that.